src/data/mv_polynomial/equiv.lean

```diff
@@ -319,6 +321,265 @@ begin
   simp only [mv_polynomial.fin_succ_equiv_apply, mv_polynomial.eval₂_hom_C],
 end
 
+lemma fin_succ_equiv_X_eq_zero {n : ℕ} {R : Type u} [comm_semiring R] :
+  fin_succ_equiv R n (X 0) = polynomial.X := by simp
+
+lemma fin_succ_equiv_X_ne_zero {n : ℕ} {R : Type u} [comm_semiring R] {j : fin n} :
+  fin_succ_equiv R n (X j.succ) = polynomial.C (X j) := by simp
+
+private lemma fin_succ_equiv_coeff_coeff_C {n : ℕ } {R : Type u} [comm_semiring R]
```

Comment:
Does it make sense to extract these assumptions `n R [comm_semiring R]` etc... into a `variables` statement? You are repeating them all the time.

Reply:
I now did this with `R` and `[comm_semiring R]` but not with `n` as L52 declares `n : σ`. The existing declarations on `fin_succ_equiv` have `n` as an argument.